src/Dapr.Jobs/DaprJobsGrpcClient.cs

```diff
@@ -0,0 +1,312 @@
+    /// <param name="payload">The main payload of the job.</param>
+    /// <param name="cancellationToken">Cancellation token.</param>
+    [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+    public override async Task ScheduleIntervalJobAsync(string jobName, TimeSpan interval,
+        DateTime? startingFrom = null, int? repeats = null,
+        DateTime? ttl = null, ReadOnlyMemory<byte>? payload = null, CancellationToken cancellationToken = default)
+    {
+        if (string.IsNullOrWhiteSpace(jobName))
+            throw new ArgumentNullException(nameof(jobName));
+
+        var job = new Autogenerated.Job { Name = jobName, Schedule = interval.ToDurationString() };
+
+        if (startingFrom is not null)
+            job.DueTime = ((DateTime)startingFrom).ToString("O");
+
+        if (repeats is not null)
+        {
+            if (repeats < 0)
+                throw new ArgumentOutOfRangeException(nameof(repeats));
+
+            job.Repeats = (uint)repeats;
+        }
+
+        if (payload is not null)
+            job.Data = job.Data = new Any { Value = ByteString.CopyFrom(payload.Value.Span), TypeUrl = "dapr.io/schedule/jobpayload" };
+
+        if (ttl is not null)
+        {
+            if (ttl < startingFrom)
+                throw new ArgumentException(
+                    $"When both {nameof(ttl)} and {nameof(startingFrom)} are specified, the {nameof(ttl)} must represent a later point in time");
+
+            job.Ttl = ((DateTime)ttl).ToString("O");
+        }
+        
+        var envelope = new Autogenerated.ScheduleJobRequest { Job = job};
+
+        var callOptions = CreateCallOptions(headers: null, cancellationToken);
+
+        try
+        {
+            await client.ScheduleJobAlpha1Async(envelope, callOptions);
+        }
+        catch (RpcException ex)
+        {
+            throw new DaprException(
+                "Schedule job operation failed: the Dapr endpoint indicated a failure. See InnerException for details.",
+                ex);
+        }
+    }
+
+    /// <summary>
+    /// Schedules a one-time job.
+    /// </summary>
+    /// <param name="jobName">The name of the job being scheduled.</param>
+    /// <param name="scheduledTime">The point in time when the job should be run.</param>
+    /// <param name="payload">Stores the main payload of the job which is passed to the trigger function.</param>
+    /// <param name="cancellationToken">Cancellation token.</param>
+    [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+    public override async Task ScheduleOneTimeJobAsync(string jobName, DateTime scheduledTime, ReadOnlyMemory<byte>? payload = null,
+        CancellationToken cancellationToken = default)
+    {
```

Comment:
I wonder if some of this duplicate logic could be consolidated to reduce risk of divergence.

Reply:
Done in the latest iteration